### ISSUE TITLE
modernize and revive linters in the latest 2.7.0

### ISSUE THREAD

**Issue by alex-aizman** [golangci/golangci-lint#6232]
### Welcome

- [x] Yes, I'm using a binary release within 2 latest [releases](https://github.com/golangci/golangci-lint/releases). Only such installations are supported.
- [x] Yes, I've searched similar [issues on GitHub](https://github.com/golangci/golangci-lint/issues) and didn't find any.
- [x] Yes, I've read the `typecheck` section of the [FAQ](https://golangci-lint.run/docs/welcome/faq/#why-do-you-have-typecheck-errors).
- [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).
- [x] I agree to follow this project's [Code of Conduct](https://github.com/golangci/golangci-lint?tab=coc-ov-file)

### How did you install golangci-lint?

I don't know

### Description of the problem

it is a minor problem that boils down to false positives and wrong recommendations

reproducible if you just

```sh
git clone https://github.com/NVIDIA/aistore
make lint
```

The result:

```sh
$ golangci-lint version
golangci-lint has version 2.7.0 built with go1.25.4 from e6ebea01 on 2025-12-03T19:50:52Z
$ 
$ make lint
Running lint...
ais/backend/azure.go:194:11: stringscut: strings.Index can be simplified using strings.Cut (modernize)
                if i := strings.Index(line, azErrCode); i > 0 {
                        ^
ais/dpq.go:244:10: stringscut: strings.IndexByte can be simplified using strings.Cut (modernize)
        if i := strings.IndexByte(s, '='); i > 0 {
                ^
cmn/bck.go:120:9: stringscut: strings.IndexByte can be simplified using strings.Cut (modernize)
        idx := strings.IndexByte(s, apc.NsNamePrefix)
               ^
cmn/certloader/certloader.go:314:7: stringscut: strings.Index can be simplified using strings.Cut (modernize)
        i := strings.Index(s, " +")
             ^
cmn/cos/template.go:405:12: stringscut: strings.IndexByte can be simplified using strings.Cut (modernize)
                right := strings.IndexByte(template, '@')
                         ^
cmn/k8s/k8s.go:113:9: stringscut: strings.IndexByte can be simplified using strings.Cut (modernize)
        idx := strings.IndexByte(msg, ',')
               ^
fs/fqn.go:57:7: stringscut: strings.IndexByte can be simplified using strings.Cut (modernize)
        j := strings.IndexByte(ct, filepath.Separator)
             ^
api/multiobj.go:5:9: var-naming: avoid meaningless package names (revive)
package api
        ^
8 issues:
* modernize: 7
* revive: 1
make: *** [Makefile:291: lint] Error 1
```

Problems:
1) IndexByte() does not apply
2) Index() should not be recommended when we are specifically looking for `i > 0`
3) revive error makes no sense

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
# Paste output here
```

</details>

### Configuration

<details>

```console
# paste configuration file or CLI flags here
```

</details>

### Go environment

<details>

```console
$ go version && go env
# paste output here
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
# paste output here
```

</details>

### A minimal reproducible example or link to a public repository

<details>

```go
// add your code here
```

</details>

### Validation

- [x] Yes, I've included all information above (version, config, etc.).

### Supporter

- [ ] I am a sponsor/backer through [GitHub](https://github.com/sponsors/golangci/) or [OpenCollective](https://opencollective.com/golangci-lint)

**Comment by ldez** [comment]
Hello,

this is not something specific to golangci-lint.

You checked this box:

> - [x] Yes, I've tried with the standalone [linter](https://golangci-lint.run/docs/linters/) if available (e.g., gocritic, go vet, etc.).

So, you understand that these are reports are related to revive and modernize.
Golangci-lint is wrapper around those linters.

---

```
api/multiobj.go:5:9: var-naming: avoid meaningless package names (revive)
package api
        ^
```

This is because the package is named `api`: https://github.com/mgechev/revive/blob/HEAD/RULES_DESCRIPTIONS.md#var-naming

**Comment by ldez** [comment]
About `modernize`, the reports are rights.

For example, the following function

```go
func _dpqKeqV(s string) (string, string, bool) {
	if i := strings.IndexByte(s, '='); i > 0 {
		return s[:i], s[i+1:], true
	}
	return s, "", false
}
```

can be replaced with `strings.Cut(s, "=")`


You can run `golangci-lint run --enable-only modernize --fix` to apply the suggested fixes.

I did it locally, and the fixes are right.

**Comment by alex-aizman** [comment]
not exactly, strings.Cut() is `ok` when index is zero

forces me to do not one check but two

looks clumsy and provides no value

**Comment by ldez** [comment]
I created an issue on `x/tools` https://github.com/golang/go/issues/76687

**Comment by ldez** [comment]
I will disable `stringscut` because, in the current state, it will lead to bugs.
